src/test/kotlin/no/hvl/dat251/fjelltur/service/RuleServiceImplTest.kt

```diff
@@ -220,4 +224,140 @@ internal class RuleServiceImplTest {
+  fun `update distance rule with new min kilometers`() {
+    val ruleName = "Test_distance_rule"
+    val ruleBody = "This is the body"
+    val createRequest = CreateDistanceRuleRequest(ruleName, ruleBody, 1, 10)
+    ruleService.createDistanceRule(createRequest)
+    val updatedMinKm = 2
+    val updateDistanceRuleRequest = UpdateDistanceRuleRequest(ruleName, null, null, updatedMinKm)
+    ruleService.updateDistanceRule(updateDistanceRuleRequest)
+    val updatedDistanceRule = ruleRepository.findAllByName(ruleName) as DistanceRule
+    assertEquals(updatedMinKm, updatedDistanceRule.minKilometers)
+  }
+  @WithMockUser(authorities = [CRUD_RULE_PERMISSION])
+  @Test
+  fun `update distance rule with no updated values doesn't do anything`() {
+    val ruleName = "Test_distance_rule"
+    val ruleBody = "This is the body"
+    val createRequest = CreateDistanceRuleRequest(ruleName, ruleBody, 1, 10)
+    val distanceRule = ruleService.createDistanceRule(createRequest)
+
+    val updateDistanceRuleRequest = UpdateDistanceRuleRequest(ruleName, null, null, null)
+
+    ruleService.updateDistanceRule(updateDistanceRuleRequest)
+    val updatedDistanceRule = ruleRepository.findAllByName(ruleName) as DistanceRule
+
+    assertEquals(distanceRule, updatedDistanceRule)
+  }
+
+  @WithMockUser(authorities = [CRUD_RULE_PERMISSION])
+  @Test
+  fun `update time rule that doesn't exist should throw`() {
+    val ruleName = "Test time rule"
+    val updateTimeRuleRequest = UpdateTimeRuleRequest(ruleName, null, null, null)
+
+    assertThrows<UnknownRuleNameException> { ruleService.updateTimeRule(updateTimeRuleRequest) }
+  }
+
+  @WithMockUser(authorities = [CRUD_RULE_PERMISSION])
+  @Test
+  fun `update time rule with new body`() {
+    val ruleName = "Test time rule"
+    makeNewBasicTimeRule(name = ruleName)
+
+    val newBody = "This is the new updated body"
+    val updateTimeRuleRequest = UpdateTimeRuleRequest(ruleName, newBody, null, null)
+
+    val newTimeRule = ruleService.updateTimeRule(updateTimeRuleRequest)
+
+    assertEquals(newBody, newTimeRule.body)
+  }
+
+  @WithMockUser(authorities = [CRUD_RULE_PERMISSION])
+  @Test
+  fun `update time rule with new basic points`() {
+    val ruleName = "Test time rule"
+    makeNewBasicTimeRule(name = ruleName)
+
+    val basicPoints = 1000
+    val updateTimeRuleRequest = UpdateTimeRuleRequest(ruleName, null, basicPoints, null)
+
+    val newTimeRule = ruleService.updateTimeRule(updateTimeRuleRequest)
+
+    assertEquals(basicPoints, newTimeRule.basicPoints)
+  }
+  @WithMockUser(authorities = [CRUD_RULE_PERMISSION])
+  @Test
+  fun `update time rule with new basic minMin`() {
+    val ruleName = "Test time rule"
+    makeNewBasicTimeRule(name = ruleName)
+
+    val newMinMin = 1000
+    val updateTimeRuleRequest = UpdateTimeRuleRequest(ruleName, null, null, newMinMin)
+
+    val newTimeRule = ruleService.updateTimeRule(updateTimeRuleRequest)
+
+    assertEquals(newMinMin, newTimeRule.minimumMinutes)
+  }
```

Comment:
Nydelige tester, men ser ingen tester som sjekker at feltene som er `null` i requesten ikke endres

Altså

```kotlin
val basicPoints = 1
val body = "test body"
makeNewBasicTimeRule(ruleName, body, basicPoints)

//...

assertEquals(body, newTimeRule.body)
assertEquals(basicPoints, newTimeRule.basicPoints)
```

Reply:
```
@WithMockUser(authorities = [CRUD_RULE_PERMISSION])
  @Test
  fun `update distance rule with no updated values doesn't do anything`() {
    val ruleName = "Test_distance_rule"
    val ruleBody = "This is the body"
    val createRequest = CreateDistanceRuleRequest(ruleName, ruleBody, 1, 10)
    val distanceRule = ruleService.createDistanceRule(createRequest)

    val updateDistanceRuleRequest = UpdateDistanceRuleRequest(ruleName, null, null, null)

    ruleService.updateDistanceRule(updateDistanceRuleRequest)
    val updatedDistanceRule = ruleRepository.findAllByName(ruleName) as DistanceRule

    assertEquals(distanceRule, updatedDistanceRule)
  }
```
Vil si denne gjør det for distance?

Reply:
Ja